lib/mayaUsd/ufe/UsdUndoInsertChildCommand.cpp

```diff
@@ -81,21 +85,11 @@ UsdUndoInsertChildCommand::UsdUndoInsertChildCommand(const UsdSceneItem::Ptr& pa
         _ufeDstPath = parent->path() + Ufe::PathSegment(
             Ufe::PathComponent(childName), cRtId, cSep);
     }
-    _usdDstPath = parent->prim().GetPath().AppendChild(TfToken(childName));
-
-    _childLayer = child->prim().GetStage()->GetEditTarget().GetLayer();
+    _usdDstPath = parentPrim.GetPath().AppendChild(TfToken(childName));
 
-    // If parent prim is the pseudo-root, no def primSpec will be found, so
-    // just use the edit target layer.
-    _parentLayer = parentPrim.IsPseudoRoot() 
-        ? parent->prim().GetStage()->GetEditTarget().GetLayer() 
-        : MayaUsdUtils::defPrimSpecLayer(parentPrim);
+    _childLayer = childPrim.GetStage()->GetEditTarget().GetLayer();
```

Comment:
@ppt-adsk  clean up the legacy old code for getting the appropriate layer for _parentLayer .  

ufe::applyCommandRestriction in line 71 and 72 will always put us in a position to use appropriate layer to use for performing the parent operation.